### PR TITLE
Feature/fix emtpy assembly view layers

### LINKED ISSUE #639: Layers with bricks only appear as three layers in assembly view
Because no bricks are added in these 'dead' layers everything is grayed out confusing the user.
image

These layers should be reduced to a single layer.

### PR DESCRIPTION
**Pull request by AtiX**
Closes #639 

The check with the first layer being 3d printed is a bit hacky (but already has been there before this pull request), but code beautification can/will be done when dealing with #630 since it will need a major rework of the layer-management

### CODE REVIEW

File: src/plugins/nodeVisualizer/visualization/brickVisualization.coffee [hunk 1]
```diff
@@ -185,10 +185,17 @@ class BrickVisualization
...
 
 	showAllBrickLayers: =>
-		for layer in @bricksSubnode.children
+		for layer in @_visibleChildLayers
 			layer.visible = true
 			@_makeLayerColored layer
 
+	getNumberOfVisibleLayers: =>
+		@_visibleChildLayers = @bricksSubnode.children.filter (layer) ->
+			if layer.children.length > 0
```
Comment (Steditor) [inline src/plugins/nodeVisualizer/visualization/brickVisualization.coffee]: `return layer.children.length > 0`

File: src/plugins/nodeVisualizer/nodeVisualizer.coffee [hunk 2]
```diff
@@ -304,7 +304,7 @@ class NodeVisualizer
 					@_applyStabilityView cachedData
 				when 'build'
 					@_resetStabilityView cachedData
-					@_applyBuildMode cachedData
+					return @_applyBuildMode cachedData
```
Comment (Steditor) [inline src/plugins/nodeVisualizer/nodeVisualizer.coffee]: Please don't - it is the least expected thing a visualizer could possibly return here (imho it shouldn't return anything but maybe `true/false` for success/failure), and we have access to the grid in [L40](https://github.com/brickify/brickify/pull/640/files#diff-bfac547fa50be1d57d1efb223943f44fR40).

Comment (Steditor) [inline src/plugins/nodeVisualizer/nodeVisualizer.coffee]: Or - at least get the number via a decent `getter` that describes the value that is returned.

File: src/plugins/nodeVisualizer/visualization/brickVisualization.coffee [hunk 3]
```diff
@@ -185,10 +185,15 @@ class BrickVisualization
...
-		for layer in @bricksSubnode.children
+		for layer in @_visibleChildLayers
 			layer.visible = true
 			@_makeLayerColored layer
 
+	getNumberOfVisibleLayers: =>
+		@_visibleChildLayers = @bricksSubnode.children.filter (layer) ->
+			return layer.children.length > 0
+		return @_visibleChildLayers.length
+
```
Comment (stuikomma) [inline src/plugins/nodeVisualizer/visualization/brickVisualization.coffee]: This is likely to break at some point. `@_visibleChildLayers`isn't defined unless `getNumberOfVisibleLayers` was called before.

Comment (AtiX) [inline src/plugins/nodeVisualizer/visualization/brickVisualization.coffee]: But it is always called before switching to build mode ?

**Comment by stuikomma** [comment]
:boat: but please recheck my last commit

**Comment by AtiX** [comment]
:rocket:

**Pull request merged**